test/framework/arm-test-utils.js

```diff
@@ -0,0 +1,527 @@
+ *   { name: 'AZURE_AD_TEST_PASSWORD'},
+ * ];
+ */
+function ArmTestUtils(mochaSuiteObject, testPrefix, env) {
+  this.mochaSuiteObject = mochaSuiteObject;
+  this.testPrefix = this.normalizeTestName(testPrefix);
+  this.mockServerClient;
+  this.currentTest = '';
+  this.scopeWritten;
+  //Recording info
+  this.setRecordingsDirectory(__dirname + '/../recordings/' + this.testPrefix + '/');
+  this.suiteRecordingsFile = this.getRecordingsDirectory() + 'suite.' + this.testPrefix + '.nock.js';
+  this.recordingsFile = __dirname + '/../recordings/' + this.testPrefix + '.nock.js';
+  //test modes
+  this.isMocked = !process.env.NOCK_OFF;
+  this.isRecording = process.env.AZURE_NOCK_RECORD;
+  this.isPlayback = !process.env.NOCK_OFF && !process.env.AZURE_NOCK_RECORD;
+  //authentication info
+  this.subscriptionId = process.env['AZURE_SUBSCRIPTION_ID'] || 'subscription-id';
+  this.clientId = process.env['CLIENT_ID'] || 'client-id';
+  this.domain = process.env['DOMAIN'] || 'domain';
+  this.username = process.env['USERNAME'] || 'username@example.com';
+  this.password = process.env['PASSWORD'] || 'dummypassword';
+  this.secret = process.env['APPLICATION_SECRET'] || 'dummysecret';
+  this.clientRedirectUri = process.env['CLIENT_REDIRECT_URI'] || 'clientRedirectUri';
+  this.tokenCache = new FileTokenCache(this.getTockenCacheFilePath());
+  //subscriptionId should be recorded for playback
+  env.push('AZURE_SUBSCRIPTION_ID');
+  // Normalize environment
+  this.normalizeEnvironment(env);
+  this.validateEnvironment();
+  //track & restore generated uuids to be used as part of request url, like a RBAC role assignment name
+  this.uuidsGenerated = [];
+  this.currentUuid = 0;
+  this.randomTestIdsGenerated = [];
+  this.numberOfRandomTestIdGenerated = 0;
+  //stub necessary methods if in playback mode
+  this._stubMethods();
+}
+
+_.extend(ArmTestUtils.prototype, {
+  _stubMethods: function () {
+    if (this.isPlayback) {
+      sinon.stub(msRestAzure.UserTokenCredentials.prototype, 'signRequest', function (webResource, callback) {
+        return callback(null);
+      });
+
+      sinon.stub(msRestAzure.ApplicationTokenCredentials.prototype, 'signRequest', function (webResource, callback) {
+        return callback(null);
+      });
+
+      sinon.stub(this, 'createResourcegroup', function (groupName, location, callback) {
+        return callback(null);
+      });
+
+      sinon.stub(this, 'deleteResourcegroup', function (groupName, callback) {
+        return callback(null);
+      });
+    }
+  },
+  
+  _setupResourceManagementClient: function () {
+   this.resourceManagement = new ResourceManagementClient(this.createUserCredentials(), this.subscriptionId);
+   if (this.isPlayback) {
+    this.resourceManagement.longRunningOperationRetryTimeoutInSeconds = 0;
+   }
+  },
+
+  createResourcegroup: function (groupName, location, callback) {
+    console.log('Creating Resource Group: ' + groupName);
+    //if (!this.isPlayback) {
```

Comment:
i assume you re using the "stub" option? So the commented out check will eventually be gone?